### PR TITLE
Clear FDB support 

### PR DESCRIPTION
**Pull request by ViramShah91**
FDB-Clear

### CODE REVIEW

File: src/CLI/clitree/cli-xml/mac.xml [hunk 1]
```diff
@@ -320,5 +320,79 @@ limitations under the License.
     </ACTION>
     </COMMAND>
 
+<!--====================clear mac address-table dynamic <all | vlan | address>======================-->
+
+    <COMMAND
+        name="clear mac"
+        help="clear mac"
+        >
+    <ACTION>
```
Comment (joyas-joseph) [inline src/CLI/clitree/cli-xml/mac.xml:329]: Doesnt this allow to execute "clear mac" command with no other arguments?

File: src/CLI/clitree/cli-xml/mac.xml [hunk 2]
```diff
@@ -320,5 +320,79 @@ limitations under the License.
...
+        builtin="clish_nop"
+    </ACTION>
+    </COMMAND>
+
+    <COMMAND
+        help="clear Mac address table"
+        name="clear mac address-table"
```
Comment (joyas-joseph) [inline src/CLI/clitree/cli-xml/mac.xml:335]: Would prefer to have the command first and help next.

File: src/CLI/clitree/cli-xml/mac.xml [hunk 3]
```diff
@@ -320,5 +320,79 @@ limitations under the License.
...
+        name="clear mac"
+        help="clear mac"
+        >
+    <ACTION>
+        builtin="clish_nop"
+    </ACTION>
+    </COMMAND>
+
+    <COMMAND
+        help="clear Mac address table"
```
Comment (joyas-joseph) [inline src/CLI/clitree/cli-xml/mac.xml]: Clear MAC

File: models/yang/sonic/sonic-fdb.yang [hunk 4]
```diff
@@ -15,11 +15,29 @@ module sonic-fdb {
...
+
         revision 2019-10-10 {
                 description
                         "Initial revision.";
         }
 
+	rpc clear_fdb {
+                input {
+                        leaf mac-param {
+                                type string;
```
Comment (joyas-joseph) [inline models/yang/sonic/sonic-fdb.yang]: Please do add description.

File: src/CLI/clitree/cli-xml/mac.xml (hunk 1 above)
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml:329]: Following OS10 CLI.

OS10# clear mac address-table dynamic 
% Error: The command is not completed.
OS10# 

So with "clear mac address-table dynamic all" will clear dynamic FDB entries.

File: src/CLI/clitree/cli-xml/mac.xml (hunk 2 above)
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml:335]: done

File: src/CLI/clitree/cli-xml/mac.xml (hunk 3 above)
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: done

File: models/yang/sonic/sonic-fdb.yang (hunk 4 above)
Comment (ViramShah91) [reply models/yang/sonic/sonic-fdb.yang]: done

File: models/yang/annotations/sonic-fdb-annot.yang [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    yang-version "1.1";
+
+    namespace "http://openconfig.net/yang/annotation/sfdb-annot";
+    prefix "sfdb-annot";
+
+    import sonic-extensions { prefix sonic-ext; }
+    import sonic-fdb { prefix sfdb; }
+
+    deviation /sfdb:clear_fdb {
+      deviate add {
```
Comment (jeffyin-dell) [inline models/yang/annotations/sonic-fdb-annot.yang]: Can we keep indentations at 4 spaces please?

File: models/yang/sonic/sonic-fdb.yang [hunk 6]
```diff
@@ -15,11 +15,33 @@ module sonic-fdb {
         description
                 "SONIC FDB";
 
+        revision 2019-11-07 {
```
Comment (jeffyin-dell) [inline models/yang/sonic/sonic-fdb.yang]: Same comment about keeping indentations at 4 spaces. Fix it throughout the file if you need to.

File: src/CLI/clitree/cli-xml/mac.xml [hunk 7]
```diff
@@ -32,8 +32,8 @@ limitations under the License.
 
 <!--====================show mac address-table command======================-->
     <COMMAND
-        help="Mac address table"
         name="show mac address-table"
+        help="Mac address table"
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/mac.xml]: Capitalize MAC in the help string

File: src/CLI/clitree/cli-xml/mac.xml [hunk 8]
```diff
@@ -52,8 +52,8 @@ limitations under the License.
 
 <!--====================show mac address-table address <mac-address> command ======================-->
     <COMMAND
-        help="Show mac address table address for mac-address"
         name="show mac address-table address"
+        help="Show mac address table address for mac-address"
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/mac.xml]: Capitalize MAC in the help string

Comment (jeffyin-dell) [reply src/CLI/clitree/cli-xml/mac.xml]: And throughout the rest of the code as well..

**Review by jeffyin-dell — CHANGES_REQUESTED** [review]
Just style nit-picking

File: src/CLI/clitree/cli-xml/mac.xml [hunk 9]
```diff
@@ -143,7 +143,7 @@ limitations under the License.
             >
             <PARAM
                 name="address"
-                help="Mac-address of static interface"
+                help="MAC-address of static interface"
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/mac.xml]: Sometimes the help strings say "MAC address" and other times they say "MAC-address" -- I think we don't need the hyphen. Should be fine to have a hyphen "address-table" though.

File: src/CLI/clitree/cli-xml/mac.xml [hunk 10]
```diff
@@ -238,13 +238,13 @@ limitations under the License.
...
                 name="address"
-                help="Mac address of dynamic interface"
+                help="MAC address of dynamic interface"
                 ptype="SUBCOMMAND"
                 mode="subcommand"
                 >
                 <PARAM
                     name="mac-addr"
-                    help="48 bit mac-address"
+                    help="48 bit MAC-address"
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/mac.xml]: remove hyphen

File: src/CLI/clitree/cli-xml/mac.xml [hunk 11]
```diff
@@ -364,7 +364,7 @@ limitations under the License.
...
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/mac.xml]: remove hyphen

File: src/CLI/clitree/cli-xml/mac.xml [hunk 12]
```diff
@@ -53,7 +53,7 @@ limitations under the License.
 <!--====================show mac address-table address <mac-address> command ======================-->
     <COMMAND
         name="show mac address-table address"
-        help="Show mac address table address for mac-address"
+        help="Show MAC address table address for mac-address"
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/mac.xml]: remove hyphen from "mac-address" and capitalize MAC there too

**Review by jeffyin-dell — CHANGES_REQUESTED** [review]
Last nit-pick I promise!

File: src/CLI/clitree/cli-xml/mac.xml (hunk 9 above)
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: done

File: src/CLI/clitree/cli-xml/mac.xml (hunk 10 above)
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: done

File: src/CLI/clitree/cli-xml/mac.xml (hunk 11 above)
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: done

File: src/CLI/clitree/cli-xml/mac.xml (hunk 12 above)
Comment (ViramShah91) [reply src/CLI/clitree/cli-xml/mac.xml]: done

**Review by jeffyin-dell — APPROVED** [review]

**Review by joyas-joseph — APPROVED** [review]

**Pull request merged**